### PR TITLE
Feature/div 6206 new event for solicitor to submit amended petition

### PR DESCRIPTION
**Pull request by madjava**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DIV-6206

### Change description ###
Adding new **amendPetitionForRefusalRejection** petitioner event to definition file

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: definitions/divorce/json/CaseEvent.json [hunk 1]
```diff
@@ -2703,6 +2703,17 @@
     "PostConditionState": "AmendPetition",
     "SecurityClassification": "Public"
   },
+  {
+    "LiveFrom": "29/07/2020",
+    "CaseTypeID": "DIVORCE",
+    "ID": "amendPetitionForRefusalSol",
+    "Name": "Submit amended application",
+    "Description": "Submit amended application",
+    "DisplayOrder": 3,
```
Comment (yarpo) [inline definitions/divorce/json/CaseEvent.json:2712]: are you sure this is fine? I mean don't we have more than one "3" in that list now?

Comment (madjava) [reply definitions/divorce/json/CaseEvent.json:2712]: @yarpo. Would reconfirm this is ok else update to more appropriate number. Thanks

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json [hunk 2]
```diff
@@ -54,5 +54,12 @@
     "CaseEventID": "tempEditDerivedAddresses",
     "UserRole": "caseworker-divorce-superuser",
     "CRUD": "CRU"
+  },
+  {
+    "LiveFrom": "29/07/2020",
+    "CaseTypeID": "DIVORCE",
+    "CaseEventID": "amendPetitionForRefusalRejection",
```
Comment (Dizrupto) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json:61]: I think the CaseEventID should be: amendPetitionForRefusalSol

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json [hunk 3]
```diff
@@ -54,5 +54,12 @@
...
+    "UserRole": "[PETSOLICITOR]",
+    "CRUD": "CRU"
```
Comment (Dizrupto) [inline definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json:63]: I think you can also add "R" permission to the following roles:
- caseworker-divorce-courtadmin_beta
- caseworker-divorce-superuser
- caseworker-divorce-courtadmin-la
This is a convention we've had for events, to give CRU to the role that needs to use the event, and "R" for the other roles. I can't remember exactly why, I think it was related that it helps if other roles have at least "R" role when doing statistics related type of work.

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json (hunk 2 above)
Comment (madjava) [reply definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json:61]: ouch! copied over the wrong one. Thanks

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json (hunk 3 above)
Comment (madjava) [reply definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json:63]: Thanks @Dizrupto. Had the same questions in the [Jira ticket](https://tools.hmcts.net/jira/browse/DIV-6206?focusedCommentId=795811&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-795811). I'll take this as your response.

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json (hunk 2 above)
Comment (madjava) [reply definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json]: Corretcted

File: definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json (hunk 3 above)
Comment (madjava) [reply definitions/divorce/json/AuthorisationCaseEvent/AuthorisationCaseEvent-nonprod.json:63]: Updated.

File: definitions/divorce/json/CaseEvent.json (hunk 1 above)
Comment (madjava) [reply definitions/divorce/json/CaseEvent.json:2712]: > are you sure this is fine? I mean don't we have more than one "3" in that list now?

@Dizrupto Should i be "worried" about the display index? AC's is quite about it but would be nice to use a reasonable value, thats if current in not reasonable.

Comment (madjava) [reply definitions/divorce/json/CaseEvent.json:2712]: @yarpo I can confirm that for the list of Petitioner solicitor events there is no other with DisplayOrder: 3. I believe this index is fine in this case.

Comment (Dizrupto) [reply definitions/divorce/json/CaseEvent.json:2712]: Also, I think even if some have the same index CCD can handle that.

**Comment by madjava** [comment]
New event added. Happy to close this after reviews and updates.

**Pull request merged**